### PR TITLE
VZ-4876: Add network policy for istiod access from verrazzano-managed namespaces

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

This pull request adds a network policy for istiod access from namespaces that are verrazzano-managed and istio-injection enabled.  With this network policy there is no need for the network-policy webhook which was removed.  The new network policy works for both oam and poko applications.

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/security/netpol/network_policy_test.go [hunk 1]
```diff
@@ -6,16 +6,18 @@ package netpol
...
 	"encoding/json"
 	"fmt"
-	"github.com/verrazzano/verrazzano/pkg/test/framework"
-	"github.com/verrazzano/verrazzano/pkg/test/framework/metrics"
-	"k8s.io/apimachinery/pkg/api/errors"
 	"strings"
 	"time"
 
+	"github.com/verrazzano/verrazzano/pkg/k8sutil"
+
```
Comment (davisgomes) [inline tests/e2e/security/netpol/network_policy_test.go:13]: Extra space here

File: tests/e2e/security/netpol/network_policy_test.go [hunk 2]
```diff
@@ -66,26 +70,32 @@ var _ = t.BeforeSuite(func() {
...
 		return pkg.CreateNamespace(testNamespace, nsLabels)
 	}, waitTimeout, pollingInterval).ShouldNot(BeNil())
-
 	Eventually(func() error {
 		return pkg.CreateOrUpdateResourceFromFile("testdata/security/network-policies/netpol-test.yaml")
 	}, waitTimeout, pollingInterval).ShouldNot(HaveOccurred())
 	metrics.Emit(t.Metrics.With("deployment_elapsed_time", time.Since(start).Milliseconds()))
+
+	start = time.Now()
+	pkg.DeployHelloHelidonApplication(&yamlApplier, namespace, "")
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply tests/e2e/security/netpol/network_policy_test.go:79]: Deploys the OAM version.  A test was needed for an application deployed in a namespace labeled with verrazzano-managed and istio-injection.  It does not matter which application.

File: tests/e2e/security/netpol/network_policy_test.go (hunk 1 above)
Comment (bobdonat) [reply tests/e2e/security/netpol/network_policy_test.go]: fixed

File: platform-operator/helm_config/charts/verrazzano/templates/thirdparty-networkpolicy.yaml [hunk 3]
```diff
@@ -559,6 +560,10 @@ spec:
         - port: 15012
           protocol: TCP
       from:
+        - namespaceSelector:
+            matchLabels:
+              verrazzano-managed: "true"
+              istio-injection: enabled
```
Comment (pfmackin) [inline platform-operator/helm_config/charts/verrazzano/templates/thirdparty-networkpolicy.yaml:566]: is this right, one in quotes and the other with no quotes?

File: tests/e2e/security/netpol/network_policy_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (davisgomes) [reply tests/e2e/security/netpol/network_policy_test.go:78]: As long as that's what the label selector matches on that makes sense!

File: platform-operator/helm_config/charts/verrazzano/templates/thirdparty-networkpolicy.yaml (hunk 3 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/thirdparty-networkpolicy.yaml:566]: true needs to be in quotes otherwise it is a boolean

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**